src/__tests__/FileUpload.test.tsx

```
@@ -0,0 +1,850 @@
+      await user.upload(fileInput, fileWithoutExtension);
+      
+      expect(mockOnFileSelect).toHaveBeenCalledWith([fileWithoutExtension]);
+    });
+
+    test('handles extremely long filenames', async () => {
+      const mockOnFileSelect = jest.fn();
+      const user = userEvent.setup();
+      
+      render(<FileUpload onFileSelect={mockOnFileSelect} />);
+      
+      const longName = 'a'.repeat(300) + '.txt';
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)